### ISSUE TITLE
Terms and Conditions label doesn't fit longer texts

### ISSUE THREAD

**Issue by DanieliMi** [magento/magento2#24060]
### Preconditions (*)

1. Version: 2.2.9 | 2.3.2 | 2.4-develop
2. Luma Theme

### Summary
When creating Terms and Conditions with a longer checkbox text the label and required asterisk will break into new lines. I think this is not expected behavior and looks odd.

### Steps to reproduce (*)

1. Create new Condition with 
1.1. Condition Name: Any
1.2. Status: Enabled
1.3. Show Content As: Text
1.4. Applied: Manually
1.5. Checkbox Text: A text (>= 114 chars)
1.6. Content: Any
2. Stores > Configuration > Sales > Checkout > Checkout Options > Enable Terms and Conditions: Yes
3. Flush Magento Cache
4. Go to Checkout
5. Proceed to Billing Step

### Expected result (*)

I expect the Terms and Conditions checkbox to appear and being displayed nice.
![expected](https://user-images.githubusercontent.com/31039652/62686315-c87eea00-b9c4-11e9-8821-37093b2a18df.png)

### Actual result (*)

The text is too long and due to that the label is breaking into a new line. Same with the required asterisk.
![actual](https://user-images.githubusercontent.com/31039652/62686366-e2203180-b9c4-11e9-8983-6f9903ffbe87.png)

**Comment by DanieliMi** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi, here is your [Magento instance](https://i-24060-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24060-2-3-develop.instances.magento-community.engineering/admin](https://i-24060-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by DanieliMi** [comment]
> @DanieliMi do you confirm that you were able to reproduce the issue on vanilla Magento instance following steps to reproduce?

I can verify the issue on the vanilla instance.

**Comment by tprocter-ldg** [comment]
You could try adding the following style to the checkout agreement button:

`.checkout-agreements-block .action-show {
      display: contents;
}`

**Comment by amenk** [comment]
@tprocter-ldg Awesome idea, but unfortunately not IE11 compatible. https://caniuse.com/#search=display%20contents

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19030` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by DanieliMi** [comment]
Another solution would be:
```less
.checkout-agreements-block {
        .checkout-agreements {
            margin-bottom: @indent__base;

            &.fieldset {
                > .field.choice{
                    input[type="checkbox"] {
                        vertical-align: top;
                    }

                    label.label {
                        display: inline-block;
                        max-width: 90%;
                    }
                }
            }
        }

        .action-show {
            &:extend(.abs-action-button-as-link all);
            vertical-align: baseline;
            max-width: 90%
        }
    }
}
```
But this is really not a nice and clean solution just setting the `max-width` to 90%.

> You could try adding the following style to the checkout agreement button:
> 
> `.checkout-agreements-block .action-show { display: contents; }`

Is a really nice solution if there wouldn't be the lack of supported browsers.

We tried it with `display: table` and `display: table-cell` but for this we either have to adapt the markup or move the after for the * to the `.checkout-agreement.required` which is not a nice solution after all either.

**Comment by amenk** [comment]
@magento-engcom-team Can the internal ticket status of MC-19030 be viewed in any way?  Is there an estimate for the fix?

**Comment by dynamicswebkul** [comment]
@magento i am working on it.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-36819` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi. Thank you for your report. 
The issue has been fixed in magento/magento2#29413 by @Bartlomiejsz in 2.4-develop branch
Related commit(s):
 - [5a0680cd19ec70d110ba995996f6f2785258cfb9](https://github.com/magento/magento2/commit/5a0680cd19ec70d110ba995996f6f2785258cfb9)

The fix will be available with the upcoming 2.4.1 release.
